src/Server/RestRouterHandlers/TableRestRouterHandler.cpp

```diff
@@ -152,14 +166,19 @@ String TableRestRouterHandler::executePatch(const Poco::JSON::Object::Ptr & payl
 
 String TableRestRouterHandler::executeDelete(const Poco::JSON::Object::Ptr & /*payload*/, Int32 & /*http_status*/) const
 {
+    const String & database_name = getPathParameter("database");
+    const String & table_name = getPathParameter("table");
+
+    if (!tableExists(database_name, table_name))
+    {
+        return jsonErrorResponse(fmt::format("Table {}.{} doesn't exist", database_name, table_name), ErrorCodes::UNKNOWN_TABLE);
+    }
+
     if (query_context->isDistributed() && getQueryParameter("distributed_ddl") != "false")
```

Comment:
Can we implement a `getPathParameter` function as below to simplify this checking and please update the code every place to switch to this new function when we need to bool checking

```
bool getPathParameterBool(const String & name, bool default_value=false) const
{
    const auto & value = getQueryParameter(name);
    if (value.empty())
    {
        return defalut_value;
    }
   
    if (value == "false" || value == "0")
    {
        return false;
    }
    return true;
}
```

Reply:
This will be fixed in PR #56